sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateSpecs.java

```diff
@@ -121,8 +156,23 @@ private StateSpecs() {}
    * Create a state spec that is optimized for adding values frequently, and occasionally retrieving
    * all the values that have been added.
    */
+  public static <T> StateSpec<Object, BagState<T>> bag() {
+    return bag(null);
```

Comment:
Please do edit the classes and put `@Nullable` on the fields that are now nullable.